homeassistant/components/image/__init__.py

```diff
@@ -380,3 +403,34 @@ async def handle(
     ) -> web.StreamResponse:
         """Serve image stream."""
         return await async_get_still_stream(request, image_entity)
+
+
+async def async_handle_snapshot_service(
+    image: ImageEntity, service_call: ServiceCall
+) -> None:
+    """Handle snapshot services calls."""
+    hass = image.hass
+    snapshot_file: str = service_call.data[ATTR_FILENAME]
+
+    # check if we allow to access to that file
+    if not hass.config.is_allowed_path(snapshot_file):
+        raise HomeAssistantError(
```

Comment:
Shouldn't this be `ServiceValidationError`?

https://developers.home-assistant.io/docs/core/platform/raising_exceptions/